drivers/i2c/i2c_rcar.c

```diff
@@ -0,0 +1,384 @@
+};
+
+/* Registers */
+#define RCAR_I2C_ICSCR          0x00    /* Slave Control Register */
+#define RCAR_I2C_ICMCR          0x04    /* Master Control Register */
+#define RCAR_I2C_ICSIER         0x10    /* Slave IRQ Enable */
+#define RCAR_I2C_ICMIER         0x14    /* Master IRQ Enable */
+#define RCAR_I2C_ICSSR          0x08    /* Slave Status */
+#define RCAR_I2C_ICMSR          0x0c    /* Master Status */
+#define RCAR_I2C_ICCCR          0x18    /* Clock Control Register */
+#define RCAR_I2C_ICSAR          0x1c    /* Slave Address Register */
+#define RCAR_I2C_ICMAR          0x20    /* Master Address Register */
+#define RCAR_I2C_ICRXD_ICTXD    0x24    /* Receive Transmit Data Register */
+#define RCAR_I2C_ICFBSCR        0x38    /* First Bit Setup Cycle (Gen3).*/
+#define RCAR_I2C_ICFBSCR_TCYC17 0x0f    /* 17*Tcyc */
+
+#define RCAR_I2C_ICMCR_MDBS     BIT(7)  /* Master Data Buffer Select */
+#define RCAR_I2C_ICMCR_FSCL     BIT(6)  /* Forced SCL */
+#define RCAR_I2C_ICMCR_FSDA     BIT(5)  /* Forced SDA */
+#define RCAR_I2C_ICMCR_OBPC     BIT(4)  /* Override Bus Pin Control */
+#define RCAR_I2C_ICMCR_MIE      BIT(3)  /* Master Interface Enable */
+#define RCAR_I2C_ICMCR_TSBE     BIT(2)  /* Start Byte Transmission Enable */
+#define RCAR_I2C_ICMCR_FSB      BIT(1)  /* Forced Stop onto the Bus */
+#define RCAR_I2C_ICMCR_ESG      BIT(0)  /* Enable Start Generation */
+#define RCAR_I2C_ICMCR_MASTER   (RCAR_I2C_ICMCR_MDBS | RCAR_I2C_ICMCR_MIE)
+
+/* Bits to manage ICMIER and ICMSR registers */
+#define RCAR_I2C_MNR            BIT(6)  /* Master Nack Received */
+#define RCAR_I2C_MAL            BIT(5)  /* Master Arbitration lost */
+#define RCAR_I2C_MST            BIT(4)  /* Master Stop Transmitted */
+#define RCAR_I2C_MDE            BIT(3)  /* Master Data Empty */
+#define RCAR_I2C_MDT            BIT(2)  /* Master Data Transmitted */
+#define RCAR_I2C_MDR            BIT(1)  /* Master Data Received */
+#define RCAR_I2C_MAT            BIT(0)  /* Master Address Transmitted */
+
+/* Recommended settings from official documentation */
+#define RCAR_I2C_ICCCR_CDF_100_KHZ  6
+#define RCAR_I2C_ICCCR_CDF_400_KHZ  6
+#define RCAR_I2C_ICCCR_SCGD_100_KHZ 21
+#define RCAR_I2C_ICCCR_SCGD_400_KHZ 3
+
+#define MAX_WAIT_US 100
+
+/* Helper macros for I2C */
+#define DEV_I2C_CFG(dev) \
+	((const struct i2c_rcar_cfg *)(dev)->config)
+#define DEV_I2C_DATA(dev) \
+	((struct i2c_rcar_data *)(dev)->data)
+
+static uint32_t i2c_rcar_read(const struct i2c_rcar_cfg *config,
+			      uint32_t offs)
+{
+	return sys_read32(config->reg_addr + offs);
+}
+
+static void i2c_rcar_write(const struct i2c_rcar_cfg *config,
+			   uint32_t offs, uint32_t value)
+{
+	sys_write32(value, config->reg_addr + offs);
+}
+
+static void i2c_rcar_isr(const struct device *dev)
+{
+	const struct i2c_rcar_cfg *config = DEV_I2C_CFG(dev);
+	struct i2c_rcar_data *data = DEV_I2C_DATA(dev);
+
+	if (((i2c_rcar_read(config, RCAR_I2C_ICMSR)) & data->status_mask) == data->status_mask) {
```

Comment:
this line is a bit long you can do
	if (((i2c_rcar_read(config, RCAR_I2C_ICMSR)) & data->status_mask) == data->status_mask) {
```suggestion
	if (((i2c_rcar_read(config, RCAR_I2C_ICMSR)) & data->status_mask) == 
	      data->status_mask) {
```

Reply:
Fixed, thanks